hs/CalendarResponse.hs

```diff
@@ -0,0 +1,230 @@
+    bottom = ["END:VCALENDAR"]
+
+-- ** Cookie Information
+
+-- | Obtains a list with all the information about the courses obtained from the cookies.
+getCoursesInfo :: String -> [(String, String, String)]
+getCoursesInfo courses = map courseInfo allCourses
+    where
+    courseInfo course = (course !! 0, course !! 1, course !! 2)
+    allCourses = map (splitOn "-") byCourse
+    byCourse = splitOn "_" courses
+
+-- ** Database Information
+
+-- | Pulls out the information for each course from the database.
+pullDatabase :: (String, String, String) -> IO ([Time], String, String, String)
+pullDatabase (code, sect, session) =
+    if take 1 sect == "L"
+    then getData $ returnLectureTimes (T.pack code) (T.pack sect) (T.pack session)
+    else getData $ returnTutorialTimes (T.pack code) (T.pack sect) (T.pack session)
+    where
+    getData info = do
+        courseInfo <- info 
+        return (courseInfo, code, session, sect)
+
+-- ** Event Creation
+
+-- | Generates an event for each course.
+getEvent :: String -> ([Time], String, String, String) -> [String]
+getEvent timeSystem (timeFields, code, session, sect) = concat $ eventsByCourse code session sect start end dates timeSystem
+    where
+    dataInOrder = orderData timeFields
+    start = startTime dataInOrder
+    end = endTime dataInOrder
+    dates = getDates session dataInOrder
+
+-- | Creates an event for each course.
+eventsByCourse :: String -> String -> String -> [[String]] -> [[String]] -> [[(String, String)]] -> String -> [[String]]
+eventsByCourse code session sect start end dates timeSystem = map (eventsByDay code session sect timeSystem) (zip' start end dates)
+
+-- | Join three lists together in tuples of three elements
+zip' :: [a] -> [b] -> [c] -> [(a,b,c)]
+zip' _ _ [] = []
+zip' _ [] _ = []
+zip' [] _ _ = []
+zip' (x:xs) (y2:ys) (z:zs) = (x,y2,z):zip' xs ys zs
+
+-- | Creates an event for each day
+eventsByDay :: String -> String -> String -> String -> ([String], [String], [(String, String)]) -> [String]
+eventsByDay code session sect timeSystem (start, end, dates) = if session == "Y" then year else half
+    where
+    year = halfFall ++ halfWinter
+    halfFall = eventsByTime code sect start end (dates !! 0) timeSystem
+    halfWinter = eventsByTime code sect start end (dates !! 1) timeSystem
+    half = eventsByTime code sect start end (dates !! 0) timeSystem
+
+-- | Creates an event for each start/end time.
+eventsByTime :: String -> String -> [String] -> [String] -> (String, String) -> String -> [String]
+eventsByTime code sect start end date timeSystem = 
+    if fst date == "" || snd date == "" then []
+    else map (eventsByDate code sect date timeSystem) (zip start end)
+
+-- | Generates the string that represents the event for each course.
+eventsByDate :: String -> String -> (String, String) -> String -> (String, String) -> String
+eventsByDate code sect (startDate, endDate) timeSystem (start, end) = 
+    "BEGIN:VEVENT\n" ++ "DTSTAMP:" ++ timeSystem ++ 
```

Comment:
I liked the earlier approach of storing the lines in a list of `String`. You can do that here and either return `[String]`, or use `unlines` to still return a `String`